provider/azure/environ.go

```diff
@@ -1761,8 +1767,16 @@ func (env *azureEnviron) allQueuedInstances(
 		}
 		deployment := deploymentsResult.Value()
 		deployProvisioningState := resources.ProvisioningStateNotSpecified
+		deployError := "Failed"
 		if deployment.Properties != nil {
 			deployProvisioningState = deployment.Properties.ProvisioningState
+			deployError = string(deployProvisioningState)
+			if deployment.Properties.Error != nil {
+				deployError = to.String(deployment.Properties.Error.Message)
+				if deployment.Properties.Error.Details != nil && len(*deployment.Properties.Error.Details) > 0 {
+					deployError = to.String((*deployment.Properties.Error.Details)[0].Message)
```

Comment:
why can there be more than 1 error details and we only care about the first one? this feels like we are not seeing the full error

Reply:
cause the first one is the generic "Deployment failed" and the second is the real reason, and in the cases I've seen, there's nothing else beyond that

Reply:
We could append any subsequent messages. That tends to lead to these really long chunks of text in status. I can tweak it if needed in the followup work dealing with instance type selection.